woocommerce/woocommerce-android
# Feature/6881 what is wordpress

Closes: #6881


### Description
Part of Experiment 2: Add more information about the login process in pe5sF9-6C-p2

Adds a new link to log in with WordPress email screen that redirects to web site explaining why WordPress.com is needed to log in inside WooCommerce

Additionally this PR decouples `LoginSiteAddressFragment` and `LoginEmailFragment` from WordPress login library. This enables us to add changes to the UI in these 2 screens without updating the library. 

### Testing instructions
- Open Login prologue screen and click "Continue with WordPress.com" 
- Click on link "What is WordPress.com?" and see the navigates to [link](https://woocommerce.com/document/what-is-a-wordpress-com-account/) correctly
- Check that event `unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"enter_email_address","click":"what_is_wordpress_com","is_debug":true}` with property value `"click":"what_is_wordpress_com"` is tracked in console log. 

### Images/gif

https://user-images.githubusercontent.com/2663464/177784547-489b9a48-bf79-46e2-83a0-6b838d2aae8b.mp4
